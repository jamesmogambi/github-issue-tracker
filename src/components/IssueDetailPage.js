import React from "react";
import { connect } from "react-redux";

export const IssueDetailsPage = (props) => (
  <div>
    <div className="content-container">
      {props.issue ? (
        <div>
          <h1 className="page-header__title">
            Viewing details of Issue <span>{props.issue.node.title}</span>{" "}
          </h1>
          <div
            dangerouslySetInnerHTML={{ __html: props.issue.node.bodyHTML }}
          ></div>
        </div>
      ) : (
        <div className="list-item list-item--message">
          <span>No Details </span>
        </div>
      )}
    </div>
  </div>
);

const mapStateToProps = (state, props) => {
  return {
    issue: state.issues.repositoryissues.find(
      (issue) => issue.node.id === props.match.params.id
    ),
  };
};

export default connect(mapStateToProps)(IssueDetailsPage);
