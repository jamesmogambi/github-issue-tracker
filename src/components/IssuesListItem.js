import React from "react";
import { Link } from "react-router-dom";

const IssuesListItem = ({ id, title, author, createdAt }) => (
  <Link className="list-item" to={`/issueDetails/${id}`}>
    <div>
      <img src={author.avatarUrl} alt="Avatar" className="avatar" />
      <div className="list-item__sub-title">{title}</div>
    </div>
    <span className="list-item__sub-title ">opened at {createdAt}</span>
  </Link>
);

export default IssuesListItem;
