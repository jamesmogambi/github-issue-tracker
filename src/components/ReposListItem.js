import React from "react";
import { Link } from "react-router-dom";

const ReposListItem = ({ id, name, description, repositoryOwner }) => (
  <div className="list-item">
    <div> 
      <h3 className=" list-item__title">{name}</h3>
      <span className=" list-item__sub-title">{description}</span>
      <Link to={`/issues/${name}/${repositoryOwner}`}>
        <h3 className="list-item__sub-title list-link ">View Issues</h3>
      </Link>
    </div>
    <a className="button button--link " href={`https://github.com/${repositoryOwner}/${name}/tags`} target="_blank">Tags</a>
 </div>
);

export default ReposListItem;
