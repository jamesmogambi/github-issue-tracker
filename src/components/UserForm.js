import React from "react";

export default class UserForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: "",
    };
  }
  onUserNameChange = (e) => {
    const username = e.target.value;
    this.setState(() => ({ username }));
  };

  onSubmit = (e) => {
    e.preventDefault();
    if (this.state.username) {
      this.props.onSubmit({
        username: this.state.username,
      });
    }
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <input
          type="text"
          placeholder="Github username"
          autoFocus
          className="text-input"
          value={this.state.username}
          onChange={this.onUserNameChange}
        />

        <div>
          <button className="button">View Repos</button>
        </div>
      </form>
    );
  }
}
