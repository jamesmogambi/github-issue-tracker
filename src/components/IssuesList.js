import React from "react";
import { connect } from "react-redux";
import { getRepositoryIssues } from "../actions/issues";
import IssuesListItem from "./IssuesListItem";
import selectIssues from "../selectors/issues";

// export class IssuesList extends React.Component {
//   constructor(props) {
//     super(props);
//   }
//   componentWillMount() {
//     this.props.getRepositoryIssues(this.props.params);
//   }
//   render() {
//     return (
//       <div className="content-container">
//         <h1 className="page-header__title">
//           Viewing Issues of Repository <span>{this.props.params.repoName}</span>{" "}
//         </h1>

//         <div className="list-header">
//           <div className="show-for-mobile">Issues</div>
//           <div className="show-for-desktop">Issue</div>
//           <div className="show-for-desktop">Created</div>
//         </div>
//         <div className="list-body">
//           {this.props.issues.length === 0 ? (
//             <div className="list-item list-item--message">
//               <span>No Issues for this Repo</span>
//             </div>
//           ) : (
//             this.props.issues.map(({ node }) => {
//               return <IssuesListItem key={node.id} {...node} />;
//             })
//           )}
//         </div>
//       </div>
//     );
//   }
// }

// const mapDispatchToProps = (dispatch) => ({
//   getRepositoryIssues: (params) => dispatch(getRepositoryIssues(params)),
// });

// const mapStateToProps = (state) => {
//   return {
//     issues: selectIssues(state.issues.repositoryissues, state.filters),
//   };
// };

// export default connect(mapStateToProps, mapDispatchToProps)(IssuesList);

export const IssuesList = (props) => (
  <div className="content-container">
    <h1 className="page-header__title">
      Viewing Issues of Repository <span>{props.repoName}</span>{" "}
    </h1>

    <div className="list-header">
      <div className="show-for-mobile">Issues</div>
      <div className="show-for-desktop">Issue</div>
      <div className="show-for-desktop">Created</div>
    </div>
    <div className="list-body">
      {props.issues.length === 0 ? (
        <div className="list-item list-item--message">
          <span>No Issues for this Repo</span>
        </div>
      ) : (
        props.issues.map(({ node }) => {
          return <IssuesListItem key={node.id} {...node} />;
        })
      )}
    </div>
  </div>
);

export default IssuesList;
