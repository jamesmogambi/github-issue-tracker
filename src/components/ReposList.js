import React from "react";
import { connect } from "react-redux";
import ReposListItem from "./ReposListItem";

export const ReposList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Repositories</div>
      <div className="show-for-desktop">Repository</div>
      <div className="show-for-desktop">Action</div>
    </div>
    <div className="list-body">
      {props.repos.repositories.length === 0 ? (
        <div className="list-item list-item--message">
          <span>No Repos</span>
        </div>
      ) : (
        props.repos.repositories.map((repo) => {
          return (
            <ReposListItem
              key={repo.id}
              repositoryOwner={props.repos.repositoryOwner}
              {...repo}
            />
          );
        })
      )}
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    repos: state.repos,
  };
};

export default connect(mapStateToProps)(ReposList);
