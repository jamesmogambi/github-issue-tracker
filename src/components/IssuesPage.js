import React from "react";
import { connect } from "react-redux";
import { getRepositoryIssues } from "../actions/issues";
import IssuesList from "./IssuesList";
import IssuesListFilters from "./IssuesListFilters";
import selectIssues from "../selectors/issues";

export class IssuesPage extends React.Component {
  constructor(props) {
    super(props);
  }
  componentWillMount() {
    this.props.getRepositoryIssues(this.props.match.params);
  }
  render() {
    return (
      <div>
        <IssuesListFilters />
        <IssuesList issues={this.props.issues} repoName={this.props.repoName} />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getRepositoryIssues: (params) => dispatch(getRepositoryIssues(params)),
});

const mapStateToProps = (state) => {
  return {
    issues: selectIssues(state.issues.repositoryissues, state.filters),
    repoName: state.issues.repositoryname,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(IssuesPage);
