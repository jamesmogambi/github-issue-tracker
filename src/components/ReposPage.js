import React from "react";
import { connect } from "react-redux";
import Header from "./Header";
import UserForm from "./UserForm";
import ReposList from "./ReposList";
import { getRepositories } from "../actions/repos";

export class ReposPage extends React.Component {
  onSubmit = (params) => {
    this.props.getRepositories(params);
  };
  render() {
    return (
      <div>
        <div className="content-container">
          <UserForm onSubmit={this.onSubmit} />
          <ReposList />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getRepositories: (params) => dispatch(getRepositories(params)),
});

export default connect(undefined, mapDispatchToProps)(ReposPage);
