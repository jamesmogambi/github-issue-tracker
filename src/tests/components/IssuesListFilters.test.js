import React from "react";
import { shallow } from "enzyme";
import { IssuesListFilters } from "../../components/IssuesListFilters";

let setTextFilter, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  const data = { text: "" };
  wrapper = shallow(
    <IssuesListFilters setTextFilter={setTextFilter} filters={data} />
  );
});

test("should render IssuesListFilters correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should render IssuesListFilters with data correctly", () => {
  wrapper.setProps({
    filters: { text: "automated tests" },
  });
  expect(wrapper).toMatchSnapshot();
});

test("should handle text change", () => {
  const value = "screenshots";
  wrapper.find("input").simulate("change", {
    target: { value },
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});
