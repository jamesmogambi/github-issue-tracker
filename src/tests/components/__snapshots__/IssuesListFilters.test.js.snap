// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render IssuesListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search Issue"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`should render IssuesListFilters with data correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search Issue"
        type="text"
        value="automated tests"
      />
    </div>
  </div>
</div>
`;
